src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+          MethodDeclaration method = (MethodDeclaration) member;
+          // Minimize method and then simplify variable type names
+          minimizeMethod(
+              method, cu, packageName, filePath, classpath, expectedOutput, timeoutLimit);
+          System.out.println("Minimized method " + method.getName());
+        }
+      }
+    }
+  }
+
+  /**
+   * Minimize a method by minimizing each statement in turn.
+   *
+   * @param method the method to minimize, will be modified by side effect minimization of the
+   *     method is found
+   * @param compUnit compilation unit for the Java file that we are minimizing; will be modified by
+   *     side effect
+   * @param packageName the name of the package that the Java file is in
+   * @param filePath path to the Java file that we are minimizing
+   * @param classpath classpath needed to compile and run the Java file
+   * @param expectedOutput expected standard output from running the JUnit test suite
+   * @param timeoutLimit number of seconds allowed for the whole test suite to be run
+   */
+  private static void minimizeMethod(
+      MethodDeclaration method,
+      CompilationUnit compUnit,
+      String packageName,
+      String filePath,
+      String classpath,
+      String expectedOutput,
+      int timeoutLimit) {
+    List<Statement> statements = method.getBody().getStmts();
+
+    // Map from primitive variable name to the variable's value which is found in a passing assertion.
+    Map<String, String> primitiveValues = new HashMap<String, String>();
+
+    // Iterate through the list of statements, from last to first
+    for (int i = statements.size() - 1; i >= 0; i--) {
+      Statement currStmt = statements.get(i);
+
+      // Remove the current statement. We will re-insert simplifications of it.
+      statements.remove(i);
+
+      // Obtain a list of possible replacements for the current statement.
+      List<Statement> replacements = getStatementReplacements(currStmt, primitiveValues);
+      boolean replacementFound = false;
+      for (Statement stmt : replacements) {
+        // Add replacement statement to the method's body. If stmt is null, don't add
+        // anything since null represents removal of the statement.
+        if (stmt != null) {
+          statements.add(i, stmt);
+        }
+
+        // Write, compile, and run the new Java file with the new suffix "Minimized".
+        String newFilePath = writeToFile(compUnit, filePath, "Minimized");
+        if (checkCorrectlyMinimized(
+            newFilePath, classpath, packageName, expectedOutput, timeoutLimit)) {
+          // No compilation or runtime issues, obtained output is the same as the expected output.
+          // Use simplification of this statement and continue with next statement.
+          replacementFound = true;
+          // Assertions are never simplified; if this is an assertion, it's the original statement.
+          storeValueFromAssertion(currStmt, primitiveValues);
```

Comment:
`storeValueFromAssertion` is called from a location where I suspect that it never finds an assertion.  Please add a test case that shows the substitution happening.